src/Cech_complex/include/gudhi/Cech_complex_blocker.h

```diff
@@ -35,28 +42,120 @@ namespace cech_complex {
+    Point_d c = kernel_.construct_circumcenter_d_object()(begin, end);
+    FT r = kernel_.squared_distance_d_object()(c, *begin);
+    return std::make_pair(std::move(c), std::move(r));
+  }
+  
+  
   /** \internal \brief Čech complex blocker operator() - the oracle - assigns the filtration value from the simplex
    * radius and returns if the simplex expansion must be blocked.
    *  \param[in] sh The Simplex_handle.
    *  \return true if the simplex radius is greater than the Cech_complex max_radius*/
   bool operator()(Simplex_handle sh) {
+    using Point_cloud =  std::vector<Point_d>;
     Point_cloud points;
+    
+    // for each face of simplex sh, test outsider point is indeed inside enclosing ball, if yes, take it and exit loop, otherwise, new sphere is circumsphere of all vertices
+    for (auto face : sc_ptr_->simplex_vertex_range(sh)) {
+        /////////////////////////////////////////////////////////////////////
+        
+        
+        
+        
+        ///////////////////////////////////
+    }
+    
     for (auto vertex : sc_ptr_->simplex_vertex_range(sh)) {
-      points.push_back(cc_ptr_->get_point(vertex));
+        points.push_back(cc_ptr_->get_point(vertex));
+//        points.push_back(get_point_(vertex));
 #ifdef DEBUG_TRACES
       std::clog << "#(" << vertex << ")#";
 #endif  // DEBUG_TRACES
     }
-    Filtration_value radius = Gudhi::Minimal_enclosing_ball_radius()(points);
+    // TODO to remove
+    //Filtration_value radius = Gudhi::Minimal_enclosing_ball_radius()(points);
+    // Hind: Here change the algo of the enclosing Minimal_enclosing_ball_radius
+    auto point_to_be_inserted = points.back();
+    Sphere sph = get_sphere(points.cbegin(), points.cend()-1);
+
+//     Sphere sph = get_sphere(points.cbegin(), points.cend()-1);
+    CGAL::NT_converter<FT, double> cast_to_double;
+//     CGAL::NT_converter<typename Cech_complex::Point, Point_d> cast_point_d_to_double;
+    
+    std::clog << "circumcenter: " << sph.first << ", radius: " <<  std::sqrt(cast_to_double(sph.second))<< std::endl;
+    // TODO to remove
+    // Filtration_value test = std::sqrt(CGAL::to_double(sph.second));
+
+    
+    // Check that the point to be inserted is already included in the sphere of the simplex containing the preceding points
+    // TODO instead of Euclidean_distance ; use kernel_.squared_distance_d_object()(c, *begin);
+    // Add a loop on the three faces to check sphere before computing the circumsphere
+    // Add the computed sphere as cache; a vector of spheres depending on the number of faces ?
+    //
+//     if (Gudhi::Euclidean_distance()(cast_point_d_to_double(sph.first), point_to_be_inserted) > std::sqrt(cast_to_double(sph.second)))
+//     FT r = kernel_.squared_distance_d_object()(sph.first, sph.first); //*(points.cend()-1));
+    if (kernel_.squared_distance_d_object()(sph.first, point_to_be_inserted) > sph.second)
+        sph = get_sphere(points.cbegin(), points.cend());
```

Comment:
To loop over the faces of every simplex and get the minimal enclosing ball, we could either get the faces from the simplex_tree (or from somewhere else ? - I didn't manage to find an existing function to do so in simplex_tree, is there any ? or we could define a new one), or since we move incrementally in the global loop over the vertices, we could store all the computed spheres and check whether the new vertex is inside one of them (the first computed spheres would be corresponding to each face). If not, we get the circumsphere of all the points ... I'm not sure if this isn't going to be too cumbersome ...
Any other suggestion is welcome as well !